app/__init__.py

```diff
@@ -108,6 +110,19 @@ def get_github_oauth_token():
 
     @app.context_processor
     def populate_context_variable():
-        return dict(s3_cdn=get_s3_cdn_prefix())
+        return dict(s3_cdn=get_s3_cdn_prefix(),
+                    current_user=g.current_user)
+
+    @app.before_request
+    def get_user_from_cookie():
+        token = request.cookies.get('jwt')
+        g.current_user, g.jwt_exception = None, None
+        if token:
+            try:
+                payload = JWT(app.config['API_KEY']).decode(token)
+                g.current_user = User().get_userinfo_by_id(payload['user'])
+            except Exception as e:
```

Comment:
@rufuspollock If there is a problem with jwt decode the exception will throw a 500 error.
So that is why trying to catch if any decode exception e.g token expired is there and show login button in navbar. 
In this process user can try to login again and a fresh jwt token will be assigned to the user.
And I tried to capture all exception as i do not want to have any failure here.
But `g.jwt_exception = e` is completely unnecessary